src/lexer.rs

```diff
@@ -336,9 +397,8 @@ impl<'a> Lexer<'a> {
             // Parse any digits at the end of integers, or for the non-fractional part of floats.
             raw += &self.consume_chars(|c| ('0'..='9').contains(&c));
 
-            if let Some(&('.', loc)) = self.inner.peek() {
-                self.inner.next();
-                self.last_location.end = loc.end;
+            if let Some('.') = self.inner.peek_char() {
+                self.inner.next_char();
```

Comment:
We could probably have a convenience `next_char_if` since it seems to be a common operation